drivers/sensor/adxl345/adxl345.c

```diff
@@ -17,18 +17,115 @@
 
 LOG_MODULE_REGISTER(ADXL345, CONFIG_SENSOR_LOG_LEVEL);
 
+#define ADXL345_BUS_I2C DT_ANY_INST_ON_BUS_STATUS_OKAY(i2c)
+#define ADXL345_BUS_SPI DT_ANY_INST_ON_BUS_STATUS_OKAY(spi)
+
+#if ADXL345_BUS_I2C
+static int adxl345_bus_check_i2c(const union adxl345_bus *bus)
+{
+	return device_is_ready(bus->i2c.bus) ? 0 : -ENODEV;
+}
```

Comment:
suggestion: make check calls return boolean, do integer conversion at a single point.

Reply:
Correcting to follow the comment.
Organize as a function 'bus_is_ready' that returns a bool.